src/instantiate.jl

```diff
@@ -0,0 +1,104 @@
+"""
+    struct UninstantiatedOptimizer
+        optimizer
+        params::Vector{Pair{AbstractOptimizerAttribute,<:Any}}
+    end
+
+Object representing an optimizer, created by `optimizer()` parametrized by
```

Comment:
It's ambiguous to say that it "represents" an optimizer. I would say that it groups an optimizer constructor with a set of optimizer attributes.